build-system/tasks/server-tests.js

```diff
@@ -175,39 +173,36 @@ function reportResult() {
 /**
  * Runs the test in a single input file
  *
- * @return {!ReadableStream}
+ * @param {string} inputFile
  */
-function runTest() {
-  return through.obj(async (file, enc, cb) => {
-    const inputFile = file.path;
-    const input = await getInput(inputFile);
-    const testName = getTestName(inputFile);
-    const expectedOutput = await getExpectedOutput(inputFile);
-    const extraOptions = loadOptions(inputFile);
-    const transform = await getTransform(inputFile, extraOptions);
-    const output = await getOutput(transform, input);
-    try {
-      assert.strictEqual(output, expectedOutput);
-    } catch (err) {
-      ++failed;
-      logError(testName, err);
-      cb();
-      return;
-    }
-    ++passed;
-    logWithoutTimestampLocalDev(green('✔'), 'Passed', cyan(testName));
-    cb();
-  });
+async function runTest(inputFile) {
+  const input = await getInput(inputFile);
+  const testName = getTestName(inputFile);
+  const expectedOutput = await getExpectedOutput(inputFile);
+  const extraOptions = loadOptions(inputFile);
+  const transform = await getTransform(inputFile, extraOptions);
+  const output = await getOutput(transform, input);
```

Comment:
Can let some of this file IO happen async in parallel, less thread blocking on IO

```suggestion
  const testName = getTestName(inputFile);  // This is only used in the `catch` clause
  const [input,  expectedOutput, transform] = Promise.all([
    getInput(inputFile),
    getExpectedOutput(inputFile),
    getTransform(inputFile, loadOptions(inputFile)),
  ]);
  const output = await getOutput(transform, input);
```

Reply:
Excellent idea. Used this (the `Promise.all()` needs an `await`).